base/sort.jl

```diff
@@ -1736,6 +1759,37 @@ julia> v
 """
 sort(v::AbstractVector; kws...) = sort!(copymutable(v); kws...)
 
+function sort(x::NTuple{N,T};
+              alg::Algorithm=defalg(x),
+              lt=isless,
+              by=identity,
+              rev::Union{Bool,Nothing}=nothing,
+              order::Ordering=Forward,
+              scratch::Union{Vector{T}, Nothing}=nothing) where {N,T}
+    _sort(x, alg, ord(lt,by,rev,order), (;scratch))
+end
+# Folks who want to hack internals can define a new _sort(x::NTuple, ::TheirAlg, o::Ordering)
+# or _sort(x::NTuple{N, TheirType}, ::DefaultStable, o::Ordering) where N
+function _sort(x::NTuple, a::Union{DefaultStable, DefaultUnstable}, o::Ordering, kw)
+    if length(x) > 9
```

Comment:
Would be good to avoid this magic number i.e. make it a named global and have a comment on how this specific number was chosen (based on some performance / compilation time tradeoff I guess?)

Reply:
```suggestion
    # The unrolled tuple sort is prohibitively slow to compile for length > 9.
    # See https://github.com/JuliaLang/julia/pull/46104#issuecomment-1435688502 for benchmarks
    if length(x) > 9  
```

Reply:
In my PR `Base.Any32` is used for the cutoff. It's a type constant meant for exactly this kind of usage.